### PR TITLE
chore: apn app fixes for ci

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-421](https://linear.app/customerio/issue/MBL-421/react-native-sample-apps-to-firebase-distribution)

### Changes

- Fixed APN sample by adding changes missing from V72 upgrade as suggested by [React Native upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.69.1&to=0.72.4)
- Added missing `ios/GoogleService-Info.plist` file to APN sample app
- Added `ci:install` action so both sample apps can run scripts independently and it can be used by Github action to build app
- Updated project and lock files so APN can be built using Github actions

### CODE REVIEW

File: Apps/APN/package.json
Comment (mrehan27) [inline Apps/APN/package.json:15]: Have to run with `--legacy-peer-deps` due to outdated dependencies, which can be fixed in another ticket

Comment (ami-aman) [reply Apps/APN/package.json:15]: I think it is the clipboard library that's quite outdated ! I am also working on a similar ticket, merging two similar PRs with similar intention (upgrade) might cause conflicts. 🤔

File: Apps/APN/ios/GoogleService-Info.plist
Comment (ami-aman) [inline Apps/APN/ios/GoogleService-Info.plist:1]: I don’t think we should push `GoogleService-Info.plist` to the repo. It contains sensitive information like the API key and should be added to `.gitignore`.

File: Apps/APN/package.json
Comment (mrehan27) [reply Apps/APN/package.json:15]: Thanks for the callout Aman. Without this change, we won't be able to compile APN app on CI, which means we won't be able to verify the changes for mentioned ticket. I believe the conflicts for clipboard dependency would only be in the lock file, and once they are resolved, we can remove `--legacy-peer-deps` flag, making the sample apps consistent. Resolving lock file conflicts shouldn't be an issue in my opinion. Do you see any other challenges with this?

File: Apps/APN/ios/GoogleService-Info.plist
Comment (mrehan27) [reply Apps/APN/ios/GoogleService-Info.plist:1]: `GoogleService-Info.plist` is required to build and upload sample app to Firebase, [otherwise it keeps failing](https://github.com/customerio/customerio-reactnative/actions/runs/10404077415/job/28811798606#step:17:6825).

File: Apps/APN/package.json
Comment (ami-aman) [reply Apps/APN/package.json:15]: Please disregard my previous comment. I won’t encounter conflicts until I merge all my changes into `main`, which isn’t happening anytime soon. Your changes will be merged into `main`, and any conflicts may arise when the CDP feature branch is eventually merged into main, which is expected. 👍

**Review by ami-aman — APPROVED** [review]

**Comment by ami-aman** [comment]
Approved but I would appreciate if others could also review the Android updates.

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**
